apps/frontend/src/app/CreateOrder/CreateOrder.tsx

```diff
@@ -1,51 +1,124 @@
-import React, { FC, useState } from 'react';
+import React, { ChangeEvent, FC, useCallback, useRef, useState } from 'react';
 import { useForm } from 'react-hook-form';
 import { useHistory } from 'react-router-dom';
 
 import type { CreateOrderDto, OrderModel } from '@pasnik/api/data-transfer';
 import { useOrdersFacade } from '@pasnik/orders-data-access';
+import { Alert, Button, InputAdornment, Paper, TextField } from '@mui/material';
+import { formatter, validUrl } from '../../utils';
+
 export const CreateOrder: FC = () => {
   const { register, handleSubmit } = useForm();
   const { createOrder } = useOrdersFacade();
 
+  const [restaurant, setRestaurant] = useState<string>('');
+  const [menu, setMenu] = useState<string>('');
+  const [formError, setFormError] = useState<boolean>(false);
+  const [isValidInput, setIsValidInput] = useState<boolean>(false);
+  const [deliveryPrice, setDeliveryPrice] = useState<string>('');
+
+  const deliveryRef = useRef<HTMLInputElement>(null);
+
   const [error, setError] = useState<string | null>(null);
```

Comment:
error state can be also taken from react-hook-forms, yup.js schema validation can be really useful here :)

Reply:
Okey, I made error handling by react-hook-forms but I left error state still in component for axios error handling.